### DISCUSSION TITLE
Midi note start at 0 or -2 ?

### DISCUSSION THREAD

**Discussion by Mika-42** [tttapa/Control-Surface#723] [Q&A] [answered]
Hello, I am in the process of making a Midi Fighter and I have no idea if you have programmed the notes so that they start from 0 or -2, could you tell me from which value they start (and also from which value they stop, it should be 127 but I am not sure)

thank you, have a nice day ! ^^

**Comment by tttapa** [comment, marked as answer]
Do you mean the octave? The octave starts at -1, that is the note C-1 (or `MIDI_Notes::C(-1)`) is the lowest note you can play.

https://github.com/tttapa/Control-Surface/blob/f50cee7a75a8ffc6828c5fc4f1d308f066959354/src/MIDI_Constants/Notes.hpp#L22-L37

The actual values sent over MIDI are always between 0 and 127 (inclusive). You can use these values directly, or use the `MIDI_Notes` constants.

**Reply by Mika-42** [reply]
Thanks ! That's what I wanted to know
